File: docs/components_page/components/alert/auto_dismiss.R
library(dashBootstrapComponents)
library(dashHtmlComponents)

alert <- htmlDiv(
  list(
    dbcButton("Toggle",
      id = "alert-toggle-auto", className = "me-1",
      n_clicks = 0
    ),
    htmlHr(),
    dbcAlert(
      "Hello! I am an auto-dismissing alert!",
      id = "alert-auto",
      is_open = TRUE,
      duration = 4000
    )
  )
)


app$callback(
  output("alert-auto", "is_open"),
  list(
    input("alert-toggle-auto", "n_clicks"),
    state("alert-auto", "is_open")
  ),
  function(n, is_open) {
    if (n > 0) {
      return(!is_open)
    }
    return(is_open)
  }
)

File: docs/components_page/components/alert/auto_dismiss.jl
using DashBootstrapComponents

alert = html_div([
    dbc_button("Toggle", id = "alert-toggle-auto", className = "me-1", n_clicks = 0),
    html_hr(),
    dbc_alert(
        "Hello! I am an auto-dismissing alert!",
        id = "alert-auto",
        is_open = true,
        duration = 4000,
    ),
]);

callback!(
    app,
    Output("alert-auto", "is_open"),
    Input("alert-toggle-auto", "n_clicks"),
    State("alert-auto", "is_open"),
) do n, is_open
    return n > 0 ? is_open == 0 : is_open
end;
